From: Mark Pictor <mpictor@gmail.com>
Date: Tue, 14 Jun 2011 21:26:45 -0400
Subject: [PATCH 04/82] Added CPack experimental support

Moved LICENSE and REAMDE to LICENSE.txt and README.txt since CPack
only supports html/txt/rtf files
---
 CMakeLists.txt |   15 ++++++
 LICENSE        |  151 --------------------------------------------------------
 LICENSE.txt    |  151 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 README         |   27 ----------
 README.txt     |   27 ++++++++++
 5 files changed, 193 insertions(+), 178 deletions(-)
 delete mode 100644 LICENSE
 create mode 100644 LICENSE.txt
 delete mode 100644 README
 create mode 100644 README.txt

From: tpaviot <tpaviot@gmail.com>
Date: Fri, 6 May 2011 08:56:57 +0200
Subject: [PATCH 5/82] Fixed empty package: CPACK_SET_DESTDIR has to be set to
 ON.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The 'empty package' issue has been discussed here: http://www.cmake.org/pipermail/cmake/2010-February/035171.html:

"""
By default CPack does not use the DESTDIR option during the installation
phase.  Instead it sets the CMAKE_INSTALL_PREFIX to the full path of the
temporary directory being used by CPack to stage the install package. This
can be changed by setting CPACK_SET_DESTDIR to on. If the DESTDIR option is
on, CPack will use the projects cache value for CPACK_INSTALL_PREFIX, and
set DESTDIR to the temporary staging area. This allows absolute paths to be
installed under the temporary directory. Relative paths are installed into
DESTDIR/${project's CMAKE_INSTALL_PREFIX} where DESTDIR is set to the
temporary staging area.

I think you want CPACK_SET_DESTDIR, then make sure CMAKE_INSTALL_PREFIX is
how you want it before you run cpack.
"""
---
 CMakeLists.txt |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

From: Mark Pictor <mpictor@gmail.com>
Date: Tue, 14 Jun 2011 20:45:16 -0400
Subject: [PATCH 06/82] Set CPack version variables from OCE_VERSION variables

---
 CMakeLists.txt |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

From: Mark Pictor <mpictor@gmail.com>
Date: Sat, 7 May 2011 12:47:45 -0400
Subject: [PATCH 07/82] CMake: CPack Debian package support

This is intended for quick-n-dirty developer test packages, not to replace the official (and superior) Debian tools.
---
 CMakeLists.txt |   35 +++++++++++++++++++++++++++++++++++
 1 files changed, 35 insertions(+), 0 deletions(-)

From: Mark Pictor <mpictor@gmail.com>
Date: Thu, 9 Jun 2011 19:33:34 -0400
Subject: [PATCH 08/82] RPM package support (untested)

---
 CMakeLists.txt |   30 ++++++++++++++++++++++++++++++
 1 files changed, 30 insertions(+), 0 deletions(-)

From 7d9fbf50e90423ee0e9c38e9c23b9769b4c8985d Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Fri, 5 Aug 2011 08:42:08 +0200
Subject: [PATCH] Use tabulations to indent CMake files, and remove trailing
 spaces

---
 CMakeLists.txt                     |  548 ++++++++++++++++++------------------
 ProjectConfigVersion.cmake.in      |   10 +-
 adm/cmake/DRAWEXE/CMakeLists.txt   |    6 +-
 adm/cmake/OccBuildKit.cmake        |   98 +++----
 adm/cmake/TKOpenGl/CMakeLists.txt  |    6 +-
 adm/cmake/TKService/CMakeLists.txt |    6 +-
 6 files changed, 333 insertions(+), 341 deletions(-)

From 4051d938563f43470c0b4abcbaf4d4cce12b4381 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Wed, 14 Sep 2011 19:44:57 +0200
Subject: [PATCH] Tests were not found when NOT WIN32

Use the new (CMake 2.8) add_test(NAME name COMMAND command)
signature, as explained at
  http://www.cmake.org/pipermail/cmake/2002-January/002644.html

Bump CMake requirement to 2.8.
---
 CMakeLists.txt      |   16 +++++++---------
 test/CMakeLists.txt |   11 +----------
 2 files changed, 8 insertions(+), 19 deletions(-)

Index: oce/CMakeLists.txt
===================================================================
--- oce.orig/CMakeLists.txt
+++ oce/CMakeLists.txt
@@ -945,3 +945,84 @@ IF (${PROJECT_NAME}_BUNDLE_AUTOINSTALL)
 ENDIF(${PROJECT_NAME}_BUNDLE_AUTOINSTALL)
 
 
+###############################################################################
+# OCE Packaging                                                               #
+# $make package                                                               #
+###############################################################################
+
+SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "OpenCASCADE Community Edition")
+SET(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README.txt")
+SET(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.txt")
+SET(CPACK_SET_DESTDIR "ON")
+
+SET(CPACK_PACKAGE_VERSION_MAJOR ${OCE_VERSION_MAJOR})
+SET(CPACK_PACKAGE_VERSION_MINOR ${OCE_VERSION_MINOR})
+SET(CPACK_PACKAGE_VERSION_PATCH ${OCE_VERSION_PATCH})
+SET(CPACK_PACKAGE_NAME ${PROJECT_NAME} )
+
+SET(CPACK_PACKAGE_CONTACT "OCE Developers <oce-dev@googlegroups.com>")
+
+###############################################################################
+# Debian-specific CPack variables. Intended for quick-n-dirty dev packages,   #
+# not to replace the official (and superior) debian tools.                    #
+# to enable, run cmake with -DCPACK_GENERATOR="DEB", or uncomment             #
+#set(CPACK_GENERATOR "DEB")                                                   #
+###############################################################################
+if( CPACK_GENERATOR MATCHES ".*DEB.*" )
+	if( ${CMAKE_VERSION} STREQUAL "2.8.2" )
+		# bug http://vtk.org/Bug/view.php?id=11020
+		message( WARNING "CMake/CPack version 2.8.2 will not create working .deb packages!")
+	endif()
+	set(CPACK_DEBIAN_PACKAGE_SECTION "science")
+	# debian source packages aren't yet supported by cmake, but the following list will help anyone who is trying to figure out what to apt-get install before building.
+	set(CPACK_DEBIAN_PACKAGE_BUILDS_DEPENDS "debhelper (>= 6.0.7~), quilt, cmake (>= 2.8), libtool,
+		libx11-dev, libxmu-dev, libxext-dev, tcl8.5-dev, tk8.5-dev,
+		libgl1-mesa-dev | libgl-dev, libglu1-mesa-dev | libglu-dev,
+		libftgl-dev, libgl2ps-dev")
+	set(CPACK_DEBIAN_PACKAGE_DEPENDS "libc6 (>= 2.3), libfreetype6 (>= 2.2.1), libftgl2 (>= 2.1.3~rc5),
+		libgcc1 (>= 1:4.1.1), libgl1-mesa-glx, libgl2ps0, libglu1-mesa,
+		libgomp1 (>= 4.2.1), libstdc++6 (>= 4.4.0), libx11-6, libxext6,
+		libxmu6, libxt6, tcl8.5 (>= 8.5.0), tk8.5 (>= 8.5.0)")
+	set(CPACK_DEBIAN_PACKAGE_SUGGESTS "opencascade-doc")
+	set(CPACK_DEBIAN_PACKAGE_PROVIDES "libopencascade-foundation-6.5.0, libopencascade-modeling-6.5.0,
+		libopencascade-visualization-6.5.0, libopencascade-ocaf-lite-6.5.0,
+		libopencascade-ocaf-6.5.0, opencascade-draw")
+	if( BIT EQUAL 64 )
+		set( CPACK_DEBIAN_PACKAGE_ARCHITECTURE "amd64" )
+	else()
+		set( CPACK_DEBIAN_PACKAGE_ARCHITECTURE "i386" )
+	endif()
+	string( TOLOWER "${CPACK_PACKAGE_NAME}-${OCE_VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}" CPACK_PACKAGE_FILE_NAME )
+endif()
+
+###############################################################################
+# RPM-specific CPack variables.                                               #
+# to enable, run cmake with -DCPACK_GENERATOR="RPM", or uncomment             #
+#set(CPACK_GENERATOR "RPM")                                                   #
+# For RPM, package names and versions are guesses and may need changed.       #
+#                                                                             #
+# As of cmake 2.8.4, RPM requires use of a variable that is also used for     #
+# DEB,so simultaneously building multiple package types (i.e.                 #
+# -DCPACK_GENERATOR="DEB;RPM") doesn't work well. The RPM won't end up in the #
+# correct directory, and packages specified after RPM won't be built.         #
+###############################################################################
+if( CPACK_GENERATOR MATCHES ".*RPM.*" )
+	set( CPACK_RPM_PACKAGE_REQUIRES "libc6 >= 2.3, libfreetype6 >= 2.2.1, libftgl2 >= 2.1.3, libgcc1 >= 4.1.1, libgl1-mesa-glx, libgl2ps0, libglu1-mesa, libgomp1 >= 4.2.1, libstdc++6 >= 4.4.0, libx11-6, libxext6, libxmu6, libxt6, tcl8.5 >= 8.5.0, tk8.5 >= 8.5.0")
+	set( CPACK_RPM_PACKAGE_PROVIDES "libopencascade-foundation-6.5.0, libopencascade-modeling-6.5.0, libopencascade-visualization-6.5.0, libopencascade-ocaf-lite-6.5.0, libopencascade-ocaf-6.5.0, opencascade-draw" )
+	set( CPACK_PACKAGE_RELOCATABLE "FALSE" )
+	if( BIT EQUAL 64 )
+		set( CPACK_RPM_PACKAGE_ARCHITECTURE "x86_64" )
+	else()
+		set( CPACK_RPM_PACKAGE_ARCHITECTURE "i586" )
+	endif()
+	#CPackRPM ignores CPACK_RPM_FILE_NAME. Must set two other vars instead, and they interfere
+	string( TOLOWER "${CPACK_PACKAGE_NAME}-${OCE_VERSION}_${CPACK_RPM_PACKAGE_ARCHITECTURE}.rpm" CPACK_OUTPUT_FILE_NAME )
+	if( NOT CPACK_GENERATOR STREQUAL "RPM" )
+		message( WARNING "Generating an RPM with other package type(s). Due to a bug in CPackRPM, the package will not get copied to the current directory. It will be found in _CPack_Packages/Linux/RPM, and other packages may fail to build." )
+	else()
+		#this variable is used by other packagers, don't set it unless RPM is built by itself.
+		set( CPACK_PACKAGE_FILE_NAME ${CPACK_OUTPUT_FILE_NAME} )
+	endif()
+endif()
+
+INCLUDE(CPack)
